src/main/java/com/minecolonies/coremod/entity/ai/citizen/shepherd/EntityAIWorkShepherd.java

```diff
@@ -0,0 +1,504 @@
+
+            worker.getHeldItemMainhand().damageItem(1, worker);
+
+            worker.addExperience(EXP_PER_SHEEP);
+
+            for (final ItemStack item : items)
+            {
+                worker.getInventoryCitizen().addItemStackToInventory(item);
+            }
+        }
+
+        //setDelay(40);
+
+        return SHEPHERD_DECIDE;
+    }
+
+    /**
+     * Possibly dyes a sheep based on their Worker Hut Level
+     */
+    private void dyeSheepChance(final EntitySheep sheep)
+    {
+        if (worker.getWorkBuilding() != null)
+        {
+            final int chanceToDye = worker.getWorkBuilding().getBuildingLevel();
+
+            final int rand = world.rand.nextInt(100);
+
+            if (rand <= chanceToDye)
+            {
+                final int dyeInt = world.rand.nextInt(15);
+                sheep.setFleeceColor(EnumDyeColor.byMetadata(dyeInt));
+            }
+        }
+    }
+
+    private AIState breedSheep()
+    {
+        job.removeDeadSheep();
+
+        final List<EntitySheep> sheepies = job.getSheep();
+
+        final EntitySheep sheepOne = sheepies.stream().filter(sheep -> sheep.getGrowingAge() == 0).findAny().orElse(null);
+
+        if (sheepOne == null)
+        {
+            return PREPARING;
+        }
+
+        final EntitySheep sheepTwo = sheepies.stream().filter(sheep ->
+          {
+              final float range = sheep.getDistanceToEntity(sheepOne);
+              final boolean isSheepOne = sheepOne.equals(sheep);
+              return sheep.getGrowingAge() == 0 && range <= DISTANCE_TO_BREED && !isSheepOne;
+          }
+        ).findAny().orElse(null);
+
+        if (sheepTwo == null)
+        {
+            return PREPARING;
+        }
+
+        if (!equipWheat())
+        {
+            return PREPARING;
+        }
+
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.shepherd.breedingSheep"));
+
+        breedTwoSheep(sheepOne, sheepTwo);
+
+        //setDelay(40);
+
+        return SHEPHERD_DECIDE;
+    }
+
+    private AIState butcherSheep()
+    {
+        job.removeDeadSheep();
+
+        if (sheepSize() <= 0)
+        {
+            return SHEPHERD_DECIDE;
+        }
+
+        final EntitySheep sheep = job.getSheep().stream().findFirst().orElse(null);
+
+        if (!equipAxe())
+        {
+            return PREPARING;
+        }
+
+        butcherSheep(sheep);
+
+        return SHEPHERD_BUTCHER_SHEEP;
+    }
+
+    /**
+     * Butcher a sheep.
+     */
+    private void butcherSheep(EntitySheep sheep)
+    {
+
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.shepherd.butcheringSheep"));
+
+        if (worker.getHeldItemMainhand() != null && sheep != null)
+        {
+            new DamageSource(worker.getName());
+
+            final BlockPos oldSheepLocation = sheep.getPosition();
+
+            sheep.attackEntityFrom(new DamageSource(worker.getName()), (float) 5);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)